src/rtps_network_utils.erl

```diff
@@ -17,9 +17,24 @@ flags_are_ok(Cfg) ->
         and lists:member(running, Flags)
         and not lists:member(loopback, Flags).
     
-get_local_ip() ->
+get_inet_local_ip() ->
         {ok, Interfaces} = inet:getifaddrs(),
         % getting the first interface that is at least up and running and not for loopback
         [Opts | _] = [Cfg || {_Name, Cfg} <- Interfaces, flags_are_ok(Cfg)],
         get_ipv4_from_opts(Opts).
-    
\ No newline at end of file
+
+
+get_net_local_ip() ->
+        {ok, [Interface | _ ]} = net:getifaddrs(fun
+                (#{addr := #{family := inet}, flags := Flags}) -> 
+                        lists:all(fun(E) -> lists:member(E,Flags) end,[up,running]) 
+                        and not lists:member(loopback, Flags);
+                (_) -> false
+        end),
+        maps:get(addr,maps:get(addr, Interface)).
+
+get_local_ip() ->
+        case os:type() of 
+                {unix,_} -> get_net_local_ip(); 
+                _ -> get_inet_local_ip() 
+        end.
```

Comment:
Ah and the indentation should be 4 spaces.

Reply:
> Why do we need to use net vs inet? You are introducing complexity by having now 2 different codes for 2 different OSs. Plus, the old functions seam to do the same thing while still being more readable.

Because on my Mac there are multiple interfaces and I want to get the only one that has an address belonging to the `inet` family. The module `net` allows to filter the interfaces using as condition the family, I didn't found in the module `net` this capability.

Reply:
WIth the last two commits I formatted and refactored the code